### PR TITLE
Enable power source cluster to NRF board contact sensor chef-devices

### PR DESCRIPTION
**Pull request by stingchang**
Update .zap and .matter files to current chef NRF board contact sensor devices to make

1. NRF contact sensor with rechargeable battery
2. NRF contact sensor with replaceable battery

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 1]
```diff
@@ -2803,7 +2995,7 @@
               "singleton": 0,
               "bounded": 0,
               "defaultValue": "0",
-              "reportable": 1,
+              "reportable": 0,
```
Comment (erwinpan1) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2998]: @stingchang This shouldn't be modified.

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 2]
```diff
@@ -2537,7 +2553,7 @@
               "singleton": 0,
               "bounded": 0,
               "defaultValue": "0x0",
-              "reportable": 1,
+              "reportable": 0,
```
Comment (erwinpan1) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:2556]: This should not be modified

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 3]
```diff
@@ -2847,7 +2879,7 @@
               "singleton": 0,
               "bounded": 0,
               "defaultValue": null,
-              "reportable": 1,
+              "reportable": 0,
```
Comment (erwinpan1) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:2882]: This should not be modified

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 4]
```diff
@@ -2879,7 +2911,7 @@
...
```
Comment (erwinpan1) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:2914]: This should not be modified

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 5]
```diff
@@ -3027,7 +3389,7 @@
               "singleton": 0,
               "bounded": 0,
               "defaultValue": "0",
-              "reportable": 1,
+              "reportable": 0,
```
Comment (erwinpan1) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3392]: This should not be modified

**Review by andy31415 — APPROVED** [review]

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.matter [hunk 6]
```diff
@@ -2136,28 +2139,39 @@ endpoint 1 {
     callback attribute partsList;
     callback attribute generatedCommandList;
     callback attribute acceptedCommandList;
+    callback attribute eventList;
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.matter:2142]: Please do not include eventList

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 7]
```diff
@@ -2664,6 +2712,150 @@
...
+              "name": "BatVoltage",
+              "code": 11,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int32u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2731]: What's the current voltage reported? What's the battery percentage remaining? How are those values changed?

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 8]
```diff
@@ -2664,7 +2696,151 @@
               "storageOption": "RAM",
               "singleton": 0,
               "bounded": 0,
-              "defaultValue": "",
+              "defaultValue": "Battery",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2699]: "Primary Battery"

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 9]
```diff
@@ -2664,7 +2696,151 @@
...
+              "name": "BatVoltage",
+              "code": 11,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int32u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "120",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2715]: This is in millivolts. Let's use a 2x AA battery and make this 2950 (2.95V)

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 10]
```diff
@@ -2840,6 +3016,22 @@
...
+              "included": 1,
+              "storageOption": "External",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": null,
+              "reportable": 1,
+              "minInterval": 1,
+              "maxInterval": 65534,
+              "reportableChange": 0
+            },
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:3034]: Remove this

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 11]
```diff
@@ -2574,6 +2590,22 @@
...
+              "included": 1,
+              "storageOption": "External",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": null,
+              "reportable": 1,
+              "minInterval": 1,
+              "maxInterval": 65534,
+              "reportableChange": 0
+            },
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:2608]: Remove this

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 12]
```diff
@@ -2664,7 +2696,151 @@
...
+              "name": "BatTimeRemaining",
+              "code": 13,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int32u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "10",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2747]: This is in seconds --> Let's make it large, like 2 months: 60 * 24 * 3600 == 5184000

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 13]
```diff
@@ -2664,7 +2696,151 @@
...
+              "name": "BatChargeState",
+              "code": 26,
+              "mfgCode": null,
+              "side": "server",
+              "type": "BatChargeStateEnum",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "2",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2827]:
```suggestion
              "defaultValue": "4",
```

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 14]
```diff
@@ -3008,6 +3024,320 @@
...
+              "name": "BatReplacementDescription",
+              "code": 19,
+              "mfgCode": null,
+              "side": "server",
+              "type": "char_string",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "ReplaceBattery",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3205]:
```suggestion
              "defaultValue": "2x AA/LR6 Alkaline",
```

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 15]
```diff
@@ -3008,6 +3024,320 @@
...
+              "name": "BatQuantity",
+              "code": 25,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int8u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3221]:
```suggestion
              "defaultValue": "2",
```

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 16]
```diff
@@ -3008,6 +3024,320 @@
...
+              "name": "BatVoltage",
+              "code": 11,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int32u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "120",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3093]:
```suggestion
              "defaultValue": "2950",
```

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 17]
```diff
@@ -3008,6 +3024,320 @@
...
+              "name": "BatPercentRemaining",
+              "code": 12,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int8u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "100",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3109]:
```suggestion
              "defaultValue": "98",
```

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 18]
```diff
@@ -3008,6 +3024,320 @@
...
+              "name": "BatTimeRemaining",
+              "code": 13,
+              "mfgCode": null,
+              "side": "server",
+              "type": "int32u",
+              "included": 1,
+              "storageOption": "RAM",
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "10",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3125]:
```suggestion
              "defaultValue": "52560000",
```

1 year

File: examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap [hunk 19]
```diff
@@ -3008,6 +3024,320 @@
...
+              "singleton": 0,
+              "bounded": 0,
+              "defaultValue": "1",
+              "reportable": 1,
+              "minInterval": 1,
+              "maxInterval": 65534,
+              "reportableChange": 0
+            },
+            {
+              "name": "BatReplacementDescription",
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_lFAGG1bfRO.zap:3196]: Please add the following attributes

- a `BattCommonDesignation` value of 2 (AA)
- a `BattApprovedChemistry` value of 1 (Alkaline)

File: examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap [hunk 20]
```diff
@@ -2664,7 +2696,151 @@
...
               "singleton": 0,
               "bounded": 0,
-              "defaultValue": "",
+              "defaultValue": "Primary Battery",
+              "reportable": 1,
+              "minInterval": 1,
+              "maxInterval": 65534,
+              "reportableChange": 0
+            },
+            {
```
Comment (tcarmelveilleux) [inline examples/chef/devices/rootnode_contactsensor_27f76aeaf5.zap:2705]: Please add a `BatCapacity` of `350` (350mAh)

**Review by tcarmelveilleux — APPROVED** [review]

**Pull request merged**